File: backend/app/schemas/summary_bar.py
from pydantic import BaseModel


class IndexItem(BaseModel):
    name: str
    price: float | None
    change_percent: float | None


class FxItem(BaseModel):
    pair: str
    rate: float | None
    change_percent: float | None


class SummaryBarResponse(BaseModel):
    indices: list[IndexItem]
    fx: list[FxItem]
    updated_at: str | None = None

File: frontend/src/types/summary-bar.ts
export interface IndexItem {
  name: string;
  price: number | null;
  change_percent: number | null;
}

export interface FxItem {
  pair: string;
  rate: number | null;
  change_percent: number | null;
}

export interface SummaryBarResponse {
  indices: IndexItem[];
  fx: FxItem[];
  updated_at: string | null;
}
